### PR TITLE
[EDIFICE] Mesure d'audience

### PR DESCRIPTION
**Pull request by juniorode**
# Description

Mise en place du nouveau service de mesure d'audience

## Type of change

Please check options that are relevant.

- [ ] Chore (PATCH)
- [ ] Doc (PATCH)
- [ ] Bug fix (PATCH)
- [x] New feature (MINOR)

## Which packages changed?

Please check the name of the package you changed

- [ ] admin
- [ ] app-registry
- [ ] archive
- [ ] auth
- [x] audience
- [ ] cas
- [ ] common
- [ ] communication
- [ ] conversation
- [ ] directory
- [ ] feeder
- [ ] infra
- [ ] portal
- [ ] session
- [ ] test
- [ ] tests
- [ ] timeline
- [ ] workspace

# Reminder

- Security flaws
- Performance impacts (think bulk !)
- Unit tests were replayed
- Unit tests were added and/or changed
- I have updated the reminder for the version including my modifications

- [x] All done ! :smiley:

### CODE REVIEW

File: audience/src/main/java/org/entcore/audience/view/dao/ViewDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      final JsonArray params = new JsonArray();
+      params.add(module);
+      params.add(resourceType);
+
+      StringBuilder resourceIdsPlaceholder = new StringBuilder();
+      for (String resourceId : resourceIds) {
+        resourceIdsPlaceholder.append("?,");
+        params.add(resourceId);
+      }
+      resourceIdsPlaceholder.deleteCharAt(resourceIdsPlaceholder.length() - 1);
```
Comment (nabil-mansouri) [inline audience/src/main/java/org/entcore/audience/view/dao/ViewDaoImpl.java:77]: ca vaut le coup de factoriser et/ou exploiter les helper dans entcore/commons/sql/SqlClient pour gérer les valeurs pour les requêtes de type IN?

File: audience/src/main/resources/sql/000_audience.sql [hunk 2]
```diff
@@ -0,0 +1,44 @@
+CREATE SCHEMA audience;
```
Comment (nabil-mansouri) [inline audience/src/main/resources/sql/000_audience.sql:1]: comme vu ensemble il y a quelques index qui pourraient être intéressant

**Review by nabil-mansouri — APPROVED** [review]
ok pour moi

File: audience/src/main/resources/sql/000_audience.sql (hunk 2 above)
Comment (juniorode) [reply audience/src/main/resources/sql/000_audience.sql:1]: @mariusode

**Pull request merged**